app/src/main/kotlin/com/wire/android/ui/debugscreen/DebugScreen.kt

```diff
@@ -0,0 +1,131 @@
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.wire.android.ui.common.topappbar.WireTopAppBarTitle
+import com.wire.android.ui.common.topappbar.wireTopAppBarColors
+import com.wire.android.ui.home.conversationslist.common.FolderHeader
+import com.wire.android.ui.theme.wireDimensions
+import com.wire.android.ui.theme.wireTypography
+import com.wire.android.util.startFileShareIntent
+
+@Composable
+fun DebugScreen() {
+    val debugScreenViewModel: DebugScreenViewModel = hiltViewModel()
+    Column {
+        TopBar(title = "Debug")
+        ListWithHeader("MLS Data") { debugScreenViewModel.mlsData.map { TextRowItem(it) } }
+        ListWithHeader("Logs") { LoggingSection(debugScreenViewModel) }
+    }
+}
+
+@Composable
+fun TopBar(title: String) {
+    val colors = wireTopAppBarColors()
+    Surface(
+        shadowElevation = MaterialTheme.wireDimensions.topBarShadowElevation,
+        color = colors.containerColor(scrollFraction = 0f).value
+    ) {
+        CenterAlignedTopAppBar(
+            title = { WireTopAppBarTitle(title = title, MaterialTheme.wireTypography.title01) },
+            colors = colors,
+        )
+    }
+}
+
+@Composable
+fun ListWithHeader(
+    headerTitle: String, content: @Composable () -> Unit = {}
+) {
+    Column {
+        FolderHeader(headerTitle)
+        content()
+    }
+}
+
+@Composable
+fun TextRowItem(text: String, @DrawableRes trailingIcon: Int? = null, onIconClick: () -> Unit = {}) {
+    Row(modifier = Modifier.fillMaxWidth().background(Color.White)) {
+        Text(
+            text = text,
+            fontWeight = FontWeight.Normal,
+            color = MaterialTheme.colorScheme.onSecondary,
+            modifier = Modifier.padding(12.dp).weight(1f),
+            textAlign = TextAlign.Left,
+            fontSize = 14.sp
+        )
+        trailingIcon?.let {
+            Icon(painter = painterResource(id = trailingIcon),
+                contentDescription = "",
+                tint = MaterialTheme.colorScheme.onSecondary,
+                modifier = Modifier.defaultMinSize(80.dp).clickable { onIconClick() }
+            )
+        }
+
+    }
+}
+
+@Composable
+fun LoggingSection(debugScreenViewModel: DebugScreenViewModel) {
+    val context = LocalContext.current
+    val absolutePath = context.cacheDir.absolutePath
+    SwitchRowItem(
+        text = "Enable Logging", checked = debugScreenViewModel.checkedState.value
+    ) { state: Boolean ->
+        debugScreenViewModel.setLoggingEnabledState(state, absolutePath)
+    }
+    TextRowItem(
+        "Share Logs",
+        trailingIcon = android.R.drawable.ic_menu_share
+    ) { context.startFileShareIntent(debugScreenViewModel.logFilePath(absolutePath)) }
+}
+
+@Composable
+fun SwitchRowItem(
+    text: String, checked: Boolean = false, onCheckedChange: ((Boolean) -> Unit)?
+) {
+    Row(modifier = Modifier.fillMaxWidth().background(Color.White)) {
```

Comment:
Can you also change the background color to be for instance `MaterialTheme.colorScheme.surface`? 😃